pkg/manager/http.go

```diff
@@ -734,6 +739,48 @@ func (serv *HTTPServer) modulesInfo(w http.ResponseWriter, r *http.Request) {
 	serv.jsonPage(w, r, "modules", cover.Modules)
 }
 
+func (serv *HTTPServer) httpAddInput(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "multipart/form-data")
+	if r.Method != http.MethodPost {
+		http.Error(w, "only POST method supported", http.StatusMethodNotAllowed)
+		return
+	}
+	err := r.ParseMultipartForm(20 << 20)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to parse form: %v", err), http.StatusBadRequest)
+		return
+	}
+	file, _, err := r.FormFile("file")
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to retrieve file from form-data: %v", err), http.StatusBadRequest)
+		return
+	}
+	defer file.Close()
+	scanner := bufio.NewScanner(file)
+	var progData []byte
+	for scanner.Scan() {
+		data := scanner.Bytes()
+		progData = append(progData, data...)
+	}
+	target, err := prog.GetTarget(serv.Cfg.TargetOS, serv.Cfg.TargetArch)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to find target: %v", err), http.StatusBadRequest)
+		return
+	}
+	p, err := target.Deserialize(progData, prog.NonStrict)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to deserialize prog: %v", err), http.StatusBadRequest)
+		return
+	}
+	data := p.Serialize()
+	sig := hash.String(data)
+	if serv.CorpusDB != nil {
+		serv.CorpusDB.Save(sig, data, 0)
+		serv.CorpusDB.Flush()
+	}
+	w.Write([]byte("File uploaded successfully\n"))
```

Comment:
It would be useful to include result of `p.Serialize` here. Since you used `prog.NonStrict` the resulting program may be completely different from what was uploaded.